eth-account-abstraction/src/utils.sw

```diff
@@ -0,0 +1,56 @@
+library utils;
+
+use std::{
+    b512::B512,
+    ecr::{ec_recover, EcRecoverError},
+    hash::{
+        keccak256,
+        sha256,
+    },
+    identity::Identity,
+    vm::evm::evm_address::EvmAddress,
+};
+
+//Applies the prefix used by Geth to a message hash.
+//Returns the prefixed hash.
+pub fn eth_prefix(msg_hash: b256) -> b256 {
+    let prefix = "\x19Ethereum Signed Message:\n32";
+    sha256((prefix, msg_hash))
```

Comment:
Not sure this will work correctly. See https://github.com/FuelLabs/sway/issues/2594#issuecomment-1221205410 for more context. It might be worth adding a bytes library as a requirement to implement this correctly.

Reply:
Ah, I think I see; so the library would need to give an unpadded reference type for correct hashing with sha256?